src/pykeen/nn/node_piece/representations.py

```diff
@@ -285,39 +283,12 @@ def __init__(
             )
         ]
 
-        # determine shape
-        if shape is None:
-            shapes = {t.vocabulary.shape for t in token_representations}
-            if len(shapes) != 1:
-                raise ValueError(f"Inconsistent token shapes: {shapes}")
-            shape = list(shapes)[0]
-
-        # super init; has to happen *before* any parameter or buffer is assigned
-        super().__init__(max_id=triples_factory.num_entities, shape=shape, **kwargs)
-
-        # assign module
-        self.token_representations = torch.nn.ModuleList(token_representations)
-
-        # Assign default aggregation
-        self.aggregation = aggregation_resolver.lookup(aggregation)
-        self.aggregation_index = -(1 + len(shape))
-
-    # docstr-coverage: inherited
-    def extra_repr(self) -> str:  # noqa: D102
-        aggregation_str = self.aggregation.__name__ if hasattr(self.aggregation, "__name__") else str(self.aggregation)
-        return f"aggregation={aggregation_str}, "
-
-    # docstr-coverage: inherited
-    def _plain_forward(
-        self,
-        indices: Optional[torch.LongTensor] = None,
-    ) -> torch.FloatTensor:  # noqa: D102
-        return self.aggregation(
-            torch.cat(
-                [tokenization(indices=indices) for tokenization in self.token_representations],
-                dim=self.aggregation_index,
-            ),
-            self.aggregation_index,
+        super().__init__(
+            max_id=triples_factory.num_entities,
+            base=token_representations,
+            combination=ConcatAggregationCombination,
+            combination_kwargs=dict(aggregation=aggregation, dim=-len(token_representations[0].shape)),
```

Comment:
maybe still worth saving aggregation index as a variable first to give  it a name

Reply:
This is available as the `dim` attribute of the combination.

It is now also included in the output of `repr(instance)`

e.g. for NodePieceRepresentation
```
NodePieceRepresentation(
  max_id=14, shape=(3,), unique=True
  (combination): ConcatAggregationCombination(dim=-2)
)
```

Reply:
https://github.com/pykeen/pykeen/pull/964/commits/8844be460cb6b571a4071aee4f22cde5c612afff